au/code/au/math.hh

```diff
@@ -312,6 +312,29 @@ constexpr bool isnan(QuantityPoint<U, R> p) {
     return std::isnan(p.in(U{}));
 }
 
+//
+// Linear interpolation between two values of the same dimension, as per `std::lerp`.
+//
+// Note that `std::lerp` is not defined until C++20, so neither is `au::lerp`.
+//
+// Note, too, that the implementation for same-type `Quantity` instances lives inside of the
+// `Quantity` class implementation as a hidden friend, so that we can support shapeshifter types
+// such as `Zero` or `Constant<U>`.
+//
+#if defined(__cpp_lib_interpolate) && __cpp_lib_interpolate >= 201902L
+template <typename U1, typename R1, typename U2, typename R2, typename T>
+constexpr auto lerp(Quantity<U1, R1> q1, Quantity<U2, R2> q2, T t) {
```

Comment:
Is passing in a raw scalar for `T` intended? What about forcing `Unos` so that percentages are supported?

Reply:
It was intended, because the semantics of the third argument are always dimensionless, and I expect people would use raw numbers almost always.  That said, it's a good suggestion and was something I didn't think of, and I think we could support `percent` without undue difficulty.  Check out the latest commit.